lib/features/context-pad/ContextPad.js

```diff
@@ -513,6 +516,8 @@ ContextPad.prototype.isShown = function() {
  */
 ContextPad.prototype._getPosition = function(target) {
 
+  target = isConnection(target) ? getLastWaypoint(target) : target;
+
   var elements = isArray(target) ? target : [ target ];
```

Comment:
Could we document what we're doing here? I think this is going to be confusing, especially mixed up with the `array vs. no array` logic.

Reply:
I added some phrases to the JSDocs: https://github.com/bpmn-io/diagram-js/pull/826/files#diff-6d674faf358204aeb4b308e49584a5b425dd2b59dcef70ebfb108f894b96db5dR513-R523. Is this what you had in mind?

Reply:
I think that makes stuff a lot clearer, thanks!